ament_cpplint/ament_cpplint/main.py

```diff
@@ -144,27 +148,29 @@ def main(argv=sys.argv[1:]):
         else:
             print("Not using '--root'")
         print('')
-        arguments += files
-        filenames = ParseArguments(arguments)
-
-        for filename in filenames:
-            # hook into error reporting
-            errors = []
-
-            def custom_error(filename, linenum, category, confidence, message):
-                if ament_cpplint.cpplint._ShouldPrintError(category, confidence, linenum):
-                    errors.append({
-                        'linenum': linenum,
-                        'category': category,
-                        'confidence': confidence,
-                        'message': message,
-                    })
-                DefaultError(filename, linenum, category, confidence, message)
-            ament_cpplint.cpplint.Error = custom_error
-
-            ProcessFile(filename, _cpplint_state.verbose_level)
-            report.append((filename, errors))
-            print('')
+        files_check = [f for f in files if not any(f for excl in args.excludes if excl in f)]
```

Comment:
As commented multiple times before I don't see how `excl in f` is the appropriate condition here.

Reply:
- The variables files contains all the files to check by cpplint. 
 - In args.excludes I have the excluded files or directories. 

Then I check if there is one of the excludes in the file with `any(f for excl in args.excludes if excl in f)` this will return the opposite condition that we want. That's why I negate this with the **not**. 

example: 

files = [`foo/file.h`, `foo/file2.h`, `bar/file.h`]
excludes = ['bar']

any(bar in `foo/file.h`) -> False -> not any() -> True -> we will check this file
any(bar in `foo/file2.h`) -> False -> not any() -> True -> we will check this file
any(bar in `bar/file.h`) -> True -> not any() -> False -> we will not check this file

Reply:
I havfiles = ['foo/bar/baz']e given you examples multiple time why this *doesn't* work. Just try it with:

```
files = ['foo/bar/baz']
args.excludes = ['bar']
```

The result of `files_check` is empty which is incorrect.